editor/plugins/sprite_frames_editor_plugin.cpp

```diff
@@ -533,12 +563,28 @@ void SpriteFramesEditor::_prepare_sprite_sheet(const String &p_file) {
 		split_sheet_sep_y->set_value(0);
 		split_sheet_offset_x->set_value(0);
 		split_sheet_offset_y->set_value(0);
-		updating_split_settings = false;
+	} else {
+		dominant_param = int(EditorSettings::get_singleton()->get_project_metadata("sprite_frames", "dominant_param", -1));
+		if (dominant_param == PARAM_FRAME_COUNT) {
+			split_sheet_size_x->set_value(split_data["dom_x"]);
+			split_sheet_size_y->set_value(split_data["dom_y"]);
+		} else {
+			split_sheet_h->set_value(split_data["dom_x"]);
+			split_sheet_v->set_value(split_data["dom_y"]);
+		}
 
-		// Reset zoom.
-		_sheet_zoom_reset();
+		split_sheet_sep_x->set_value(split_data["sep_x"]);
+		split_sheet_sep_y->set_value(split_data["sep_y"]);
+		split_sheet_offset_x->set_value(split_data["off_x"]);
+		split_sheet_offset_y->set_value(split_data["off_y"]);
 	}
 
+	updating_split_settings = false;
+
+	// Reset zoom.
```

Comment:
If it's done for a specific purpose, this comment would need to be clarified. Otherwise you can remove it, as it's obvious from the method name.